mlrun/projects/project.py

```diff
@@ -2042,11 +2043,31 @@ def _resolved_function(
         # if the name doesn't contain the tag (or was not specified) we update both the tagged and untagged entries
         # for consistency
         name = name or resolved_function_name
-        if tag and not name.endswith(f":{tag}"):
-            self.spec.set_function(f"{name}:{tag}", function_object, func)
 
-        self.spec.set_function(name, function_object, func)
-        return name, function_object, func
+        return (
+            name,
+            tag,
+            function_object,
+            func,
+        )
+
+    def _set_function(
+        self,
+        resolved_function_name: str,
```

Comment:
resoved function name doesn't really have any meaning in the context of `_set_function`
```suggestion
        name: str,
```